### PR TITLE
Fix Servlet instrumentation for servlet-api 3.1+ - remove output stream wrapper

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Fixes https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1083

The third time's the charm, I hope. This PR contains removal of the `ServletoutputStream` wrapper according to [this comment](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1083#issuecomment-685679223).

This might be not so bad anyway, because according to `ServletResponse#getOutputStream()` JavaDoc:
```
     * Either this method or {@link #getWriter} may 
     * be called to write the body, not both.
```
so only those implementations that use `getOutputStream()` will get inaccurate content sizes (always 0), while those using `getWriter()` will be perfectly fine.

### CODE REVIEW

File: instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy [hunk 1]
```diff
@@ -122,7 +122,7 @@ class DropwizardTest extends HttpServerTest<DropwizardTestSupport> {
         "${SemanticAttributes.HTTP_USER_AGENT.key()}" TEST_USER_AGENT
         "${SemanticAttributes.HTTP_CLIENT_IP.key()}" TEST_CLIENT_IP
         // exception bodies are not yet recorded
-        "${SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH.key()}" { it == responseContentLength || /* async */it == null }
+        "${SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH.key()}" { it == 0 || it == responseContentLength || /* async */it == null }
```
Comment (FrankSpitulski) [inline instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy:125]: allowing 0 indicates that the response was not counted. if it's not counted in the tests I wouldn't expect it to be counted elsewhere.

Comment (mateuszrzeszutek) [reply instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy]: Well, it's to be expected that all frameworks that use `getOutputStream()` will now get a 0.

Comment (FrankSpitulski) [reply instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy]: This means it misses spring boot's default configuration which I'd hazard to say is a large portion of applications being instrumented. Is there some alternative implementation or is this attribute just being dropped?

**Comment by FrankSpitulski** [comment]
Can you add a unit test that succeeds now but if the wrapped response were to be used would fail so that there will be no regressions?

File: instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy]: There were two:
* https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1104 contained a separate instrumentation for servlet 3.0 and 3.1; it had lots of copy-pasted code
* In https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1149 I initially tried to just copy the `WriteListener` class that was absent in 3.0, but that failed. Then I tried to generate a class extending `ServletOutputStream` in runtime, which probably should work -- but there's no way to load it correctly without modifying `Instrumenter.Default` heavily.

**Comment by mateuszrzeszutek** [comment]
> Can you add a unit test that succeeds now but if the wrapped response were to be used would fail so that there will be no regressions?

I can do a simple `instanceof` check -- the same way as it was handled in `JettyHandlerTest` for a [similar issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1156).
Calling either `ServletOutputStream#isReady()` or ` ServletOutputStream#setWriteListener()` would be ideal, but unfortunately I cannot do that without bumping servlet api version.

File: instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy (hunk 1 above)
Comment (anuraaga) [reply instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy]: @FrankSpitulski We're leaning towards just using `content-length` header for servlet. The counting approach seems to have too many corner cases for us to deal with well unfortunately 😿 

@mateuszrzeszutek I don't think we should set `0` if we couldn't count, `0` means no content, not content couldn't be counted - can you update the implementation of `getContentLength` to avoid setting `0` when `printWriter` is `null`?

**Comment by iNikem** [comment]
Taking into account https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1096#issuecomment-685450841 and https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1083#issuecomment-685679223 I propose to completely remove generic response counting support from servlet instrumentation. I feel bad about this, but if we are left with a partial solution (as @FrankSpitulski noticed above "it misses spring boot's default configuration"), then added complexity is not worth it. History has shown, that this is not as easy as we would like :(

Unless somebody has a good proposal how to solve this otherwise, I ask @mateuszrzeszutek to remove `CountingHttpServletRequest`, `CountingHttpServletResponse` and all related functionality for the time being.

**Comment by mateuszrzeszutek** [comment]
OK, I've removed all `Counting*` classes.

**Review by anuraaga — APPROVED** [review]
Thanks @mateuszrzeszutek!

@FrankSpitulski Really sorry about this >< Thanks a lot for the help and giving us the opportunity to think through some of the problems regards to instrumentation here, we may not have noticed that Muzzle needs additional checks related to base-class matching without this exercise so will hopefully get better.

**Review by iNikem — APPROVED** [review]

File: instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/dropwizard-testing/src/test/groovy/DropwizardTest.groovy]: I've removed all the counting stuff altogether, according to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1167#issuecomment-689345608

**Comment by mateuszrzeszutek** [comment]
> we may not have noticed that Muzzle needs additional checks related to base-class matching without this exercise so will hopefully get better.

There's an issue for that now: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1123 . I have an idea how to do implement that, so hopefully this will get done soon and we'll avoid similar issues in the future.
